src/Network/HTTPException/BadGatewayException.php

```diff
@@ -25,5 +25,7 @@
 
 class BadGatewayException extends HTTPException
 {
-	protected $code = 502;
+	protected $code  = 502;
+	var $httpdesc    = 'Bad Gateway';
+	var $explanation = 'The server was acting as a gateway or proxy and received an invalid response from the upstream server.';
```

Comment:
Please mention the variable scope. If it isn't meant to be modified, please create it `protected` and create an accessor method in `HTTPException` like the base `Exception->getCode` and `Exception->getMessage`.